### PR TITLE
thrift_proxy: Handle thrift void response as a success

### PR DESCRIPTION
**Pull request by fishcakez**
Commit Message: Handle thrift void response as a success
Additional Description: When a successful thrift response was received we never called fieldBegin for void response because fieldBegin is skipped if the field type is stop. Therefore check at the message end if we never saw a field begin callback so we can mark a void response as a success.
Risk Level: low
Testing: bazel test test/extensions/filters/network/thrift_proxy/...
Docs Changes: n/a
Release Notes: n/a

Signed-off-by: James Fish <jfish@pinterest.com>

### CODE REVIEW

File: source/extensions/filters/network/thrift_proxy/conn_manager.cc [hunk 1]
```diff
@@ -201,13 +201,29 @@ FilterStatus ConnectionManager::ResponseDecoder::fieldBegin(absl::string_view na
     // Reply messages contain a struct where field 0 is the call result and fields 1+ are
     // exceptions, if defined. At most one field may be set. Therefore, the very first field we
     // encounter in a reply is either field 0 (success) or not (IDL exception returned).
-    success_ = field_id == 0 && field_type != FieldType::Stop;
+    // If first fieldType is FieldType::Stop then it is a void success and handled in messageEnd()
+    // because decoder state machine does not call decoder event callback fieldBegin on FieldType::Stop.
+    success_ = field_id == 0;
```
Comment (fishcakez) [inline source/extensions/filters/network/thrift_proxy/conn_manager.cc:206]: @zuercher should I revert the change on this line? Given the comment I added the existing code was confusing.

**Comment by rgs1** [comment]
> Looks good to me. @rgs1 can you have a look as well?

Looking
